packages/rocketchat-lib/server/methods/updateMessage.js

```diff
@@ -0,0 +1,36 @@
+Meteor.methods({
+	updateMessage(message) {
+
+		check(message, Match.ObjectIncluding({_id:String}));
+
+		if (!Meteor.userId()) {
+			throw new Meteor.Error('error-invalid-user', 'Invalid user', { method: 'updateMessage' });
+		}
+
+		const originalMessage = RocketChat.models.Messages.findOneById(message._id);
+
+		if (!originalMessage && !originalMessage._id) {
+			return;
+		}
+
+		const hasPermission = RocketChat.authz.hasPermission(Meteor.userId(), 'edit-message', message.rid);
+		const editAllowed = RocketChat.settings.get('Message_AllowEditing');
+		const editOwn = originalMessage && originalMessage.u && (originalMessage.u._id === Meteor.userId());
```

Comment:
If originalMessage does not exist the code will return on line 13, so you don't need to check for it again.

You can use only
`const editOwn = originalMessage.u && originalMessage.u._id === Meteor.userId();`